server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregationFactory.java

```diff
@@ -0,0 +1,173 @@
+import org.opensearch.search.aggregations.support.MultiTermsValuesSourceConfig;
+import org.opensearch.search.aggregations.support.ValuesSource;
+import org.opensearch.search.aggregations.support.ValuesSourceConfig;
+import org.opensearch.search.aggregations.support.ValuesSourceRegistry;
+import org.opensearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.opensearch.search.aggregations.bucket.terms.MultiTermsAggregationBuilder.REGISTRY_KEY;
+
+/**
+ * Factory of {@link MultiTermsAggregator}.
+ */
+public class MultiTermsAggregationFactory extends AggregatorFactory {
+
+    private final List<Tuple<ValuesSourceConfig, IncludeExclude>> configs;
+    private final List<DocValueFormat> formats;
+    /**
+     * Fields inherent from Terms Aggregation Factory.
+     */
+    private final BucketOrder order;
+    private final Aggregator.SubAggCollectionMode collectMode;
+    private final TermsAggregator.BucketCountThresholds bucketCountThresholds;
+    private final boolean showTermDocCountError;
+
+    public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
+        builder.register(
+            REGISTRY_KEY,
+            org.opensearch.common.collect.List.of(CoreValuesSourceType.BYTES, CoreValuesSourceType.IP),
+            config -> {
+                final IncludeExclude.StringFilter filter = config.v2() == null
+                    ? null
+                    : config.v2().convertToStringFilter(config.v1().format());
+                return MultiTermsAggregator.InternalValuesSourceFactory.bytesValuesSource(config.v1().getValuesSource(), filter);
+            },
+            true
+        );
+
+        builder.register(REGISTRY_KEY, org.opensearch.common.collect.List.of(CoreValuesSourceType.NUMERIC), config -> {
+            ValuesSourceConfig valuesSourceConfig = config.v1();
+            IncludeExclude includeExclude = config.v2();
+            ValuesSource.Numeric valuesSource = ((ValuesSource.Numeric) valuesSourceConfig.getValuesSource());
+            IncludeExclude.LongFilter longFilter = null;
+            if (valuesSource.isFloatingPoint()) {
+                if (includeExclude != null) {
+                    longFilter = includeExclude.convertToDoubleFilter();
+                }
+                return MultiTermsAggregator.InternalValuesSourceFactory.doubleValueSource(valuesSource, longFilter);
+            } else {
+                if (includeExclude != null) {
+                    longFilter = includeExclude.convertToLongFilter(valuesSourceConfig.format());
+                }
+                return MultiTermsAggregator.InternalValuesSourceFactory.longValuesSource(valuesSource, longFilter);
+            }
+        }, true);
+
+        builder.register(
+            REGISTRY_KEY,
+            org.opensearch.common.collect.List.of(CoreValuesSourceType.BOOLEAN, CoreValuesSourceType.DATE),
```

Comment:
I think we could safely combine that with previous `CoreValuesSourceType.NUMERIC` registration: `org.opensearch.common.collect.List.of(CoreValuesSourceType.NUMERIC, CoreValuesSourceType.BOOLEAN, CoreValuesSourceType.DATE)` since it does the same thing.

Reply:
Good point! Done.